### ISSUE TITLE
VOC数据集里的白色mask部分被算作了背景

### ISSUE THREAD

**Issue by XushengLee** [YudeWang/deeplabv3plus-pytorch#42]
作者你好，如果我理解正确的话，
`sample['segmentation']`中将白色margin转换成了背景（label为0）
`sample['mask']`中存放忽略计算的mask掩模
但是在训练的时候，
使用的是`criterion = nn.CrossEntropyLoss(ignore_index=255)`
并没有使用loss.py里mask类loss funs，这样一来，相当于白色不进入计算的margin变成了背景类。
这块代码比较好修改，但是想问一下，train.py里这样写，是因为把margin看做背景效果会更好吗？为什么没有使用自己构造的loss fun呢？
感谢这份repo，帮了我很大忙~

**Comment by YudeWang** [comment]
@XushengLee 
`sample['segmentation']` 我记得没有把255部分设置成0
`sample['mask']`把255设成0是为了生成onehot类别向量，这就是我自己做实验剩下的，可以不用管它或者删掉。
`train.py`里现在这样就行不用修改，`loss.py`似乎是我在发现可以用`ignore_index`之前写的，可以忽略它。

**Comment by XushengLee** [comment]
sample['segmentation']
![image](https://user-images.githubusercontent.com/17930974/69474780-ef401f80-0dff-11ea-92c2-ebae55d831af.png)
sample['mask']
![image](https://user-images.githubusercontent.com/17930974/69474784-0bdc5780-0e00-11ea-87f9-4aad77ebcb98.png)
可以看出seg中边缘的部分被变成0了，然后在mask中单独显示这个边缘
变成0的代码应该是这里
VOCDataset.py  line 152-153
`t = sample['segmentation']
 t[t >= self.cfg.MODEL_NUM_CLASSES] = 0  # the contour valued 255 changes to 0`
这个要用的话 好改的 问题不大 我更关心的是 这个边缘（不计入loss计算的）的有无对效果有影响吗？有相关的测试吗？
比如我是这么理解的 分割任务最困难的地方就是边缘，如果不加以限制直觉上好像不太好，但VOC数据集的边缘做法道理上也算严谨。想问问你的看法。
感谢这么及时的回复 : )

**Comment by XushengLee** [comment]
@YudeWang 补充一下 这俩个图 是从VOCDataset的__getiterm__得到的 应该就是放入网络训练前的状态了 mask的1的部分比seg的大了一圈 那一圈就是原来margin的位置

**Comment by YudeWang** [comment]
@XushengLee 
我发现原来是我本地已经改成了`t = sample['segmentation'].copy()`这样就不会把255改成0了
边界这一部分确实是最困难的部分，如果有准确标注的话肯定还是用上的好，但是如果标注不准确产生噪声，不如像VOC这样把这部分忽略不计。

**Comment by XushengLee** [comment]
哦哦 谢谢回复 确实标注不对的话 确实有不好的影响
趁这个机会我还想再问一个问题 因为我之前在做医学图像的分割
就是deeplab系列的做法好像并没有设置标签的权重（比如提高对人的weight，背景保持1这样），也没有使用dice loss，而这个两个做法在医学图像分割中应用很广。
是因为deeplab已经够robust了吗？

**Comment by YudeWang** [comment]
@XushengLee 通用物体分割和医学图像分割这俩应用场景差距蛮大的，医学图像样本数量少，variance相比通用物体也小很多。这两个应用场景上各自work的方法并不一定都是通用。

**Comment by XushengLee** [comment]
thanks~
